### ISSUE TITLE
[feature request] Commit / Pull request description written by ChatGPT

### ISSUE THREAD

**Issue by leikoilja** [pwntester/octo.nvim#501]
*feature request*

It's easy to get Open AI ChatGPT api token. 
Octo.nvim already has an access to the committed/stages files and can write PR description or commit message.
It would be amazing to put those together, hit a key binding and make ChatGPT prefill PR description or commit message, just like Copilot does in Github's enterprise plan.

**Comment by leikoilja** [comment]
> IMO any SaaS integration beyond the forge (github) is not something that belongs on Octo.nvim - but you should be able to do that by integrating it yourself in a keybinding and a script?
> 
> Maybe Octo.nvim can provide a convenient hook for messages that can trigger arbitrary scripts and act on stdout?

that indeed could be a good middle-ground if octo.nvim would have  convenient hooks that other apps/scripts can interact with :)

**Comment by wd60622** [comment]
I agree with @legobeat that custom scripts and keybindings would be the way to go. 

What would ideally be exposed to take advantage of this feature? changed files? diffs? commit messages? There might be some that already exist or easily accessible with git and gh commands.

**Comment by williambdean** [comment]
Related to https://github.com/pwntester/octo.nvim/issues/1249
